{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exercises 6 â€“ answers\n",
    "\n",
    "1\\. Fetch the all the transcripts of IRAK4 using the lookup function. Fetch the cDNA sequences of all transcripts using a single POST request, and print in FASTA format."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests, sys, json\n",
    "\n",
    "def fetch_endpoint(server, request, content_type):\n",
    "\n",
    "    r = requests.get(server+request, headers={ \"Content-Type\" : content_type})\n",
    "\n",
    "    if not r.ok:\n",
    "        r.raise_for_status()\n",
    "        sys.exit()\n",
    "\n",
    "    if content_type == 'application/json':\n",
    "        return r.json()\n",
    "    else:\n",
    "        return r.text\n",
    "\n",
    "def fetch_endpoint_POST(server, request, data, content_type):\n",
    "\n",
    "    r = requests.post(server+request,\n",
    "                      headers={ \"Content-Type\" : content_type},\n",
    "                      data=data )\n",
    "\n",
    "    if not r.ok:\n",
    "        r.raise_for_status()\n",
    "        sys.exit()\n",
    "\n",
    "    if content_type == 'application/json':\n",
    "        return r.json()\n",
    "    else:\n",
    "        return r.text\n",
    "    \n",
    "# Get the gene name\n",
    "gene_name = \"IRAK4\"\n",
    "transcripts = []\n",
    "\n",
    "# define the general URL parameters\n",
    "server = \"http://rest.ensembl.org/\"\n",
    "con = \"application/json\"\n",
    "ext_get_gene = \"lookup/symbol/homo_sapiens/\" + gene_name + \"?expand=1;\"\n",
    "\n",
    "get_gene = fetch_endpoint(server, ext_get_gene, con)\n",
    "\n",
    "for transcript in get_gene['Transcript']:\n",
    "    transcripts.append(transcript['id'])\n",
    " \n",
    "data = json.dumps({ \"ids\" : transcripts })\n",
    "\n",
    "ext_sequence = '/sequence/id/'\n",
    "\n",
    "sequences = fetch_endpoint_POST(server, ext_sequence, data, con)\n",
    "\n",
    "for query in sequences:\n",
    "    print (\">\", query['id'])\n",
    "    print (query['seq'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "2\\. Get all variants that are located on chromosome 17 between 80348215 and 80348333. Get the variant class, evidence attributes, source and the most_severe_consequence for all variants in that region from the variant POST endpoint."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests, sys, json\n",
    "\n",
    "def fetch_endpoint(server, request, content_type):\n",
    "\n",
    "    r = requests.get(server+request, headers={ \"Content-Type\" : content_type})\n",
    "\n",
    "    if not r.ok:\n",
    "        r.raise_for_status()\n",
    "        sys.exit()\n",
    "\n",
    "    if content_type == 'application/json':\n",
    "        return r.json()\n",
    "    else:\n",
    "        return r.text\n",
    "\n",
    "def fetch_endpoint_POST(server, request, data, content_type):\n",
    "\n",
    "    r = requests.post(server+request,\n",
    "                      headers={ \"Content-Type\" : content_type},\n",
    "                      data=data )\n",
    "\n",
    "    if not r.ok:\n",
    "        r.raise_for_status()\n",
    "        sys.exit()\n",
    "\n",
    "    if content_type == 'application/json':\n",
    "        return r.json()\n",
    "    else:\n",
    "        return r.text\n",
    "\n",
    "con = \"application/json\"\n",
    "\n",
    "locus = \"17:80348215-80348333\"\n",
    "\n",
    "overlap_ext = \"overlap/region/human/\" + locus + \"?feature=variation;\"\n",
    "\n",
    "variants = fetch_endpoint(server, overlap_ext, con)\n",
    "variant_ids = []\n",
    "\n",
    "for variant in variants:\n",
    "    id = variant['id']\n",
    "    variant_ids.append(id)\n",
    "\n",
    "data = json.dumps({ \"ids\" : variant_ids })\n",
    "\n",
    "var_ext = \"variation/homo_sapiens\"\n",
    "\n",
    "post_variants = fetch_endpoint_POST(server, var_ext, data, con)\n",
    "\n",
    "print (\"ID\\tClass\\tEvidence\\tSource\\tMost severe consequence\")\n",
    "\n",
    "for key, value in post_variants.items():\n",
    " \n",
    "    id = post_variants[key]['name']\n",
    "    cls = post_variants[key]['var_class']\n",
    "    evidence = post_variants[key]['evidence']\n",
    "    source = post_variants[key]['source']\n",
    "    severe = post_variants[key]['most_severe_consequence']\n",
    " \n",
    "    print (id + \"\\t\" + cls + \"\\t\" + \", \".join(evidence) + \"\\t\" + source + \"\\t\" + severe)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "[Next page: Rate limiting](7_Rate_limiting.ipynb)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
