---
title: "R Notebook"
output: html_notebook
---

# Follows from: orthology_maps_data_inputs.Rmd.

# (Can) work in parallel with: ciona_larvae_clustering.Rmd.

# General Description: Maps all cell types using Wagner's DEGs in zeb_integrated_50.

# Note: This script is used for visualization for Tissue Mapping only! 
# ClusterIDs are present in zeb_degs_heatmaps.Rmd.

# Env used: env_zeb_24hpf_clustering.RData.

# Complete Pipeline:

# (1) Makes zeb_*t2_24hpf_seurat (6 of them, _t5). These are unscaled.
# (2) Uses above to make various integrated objects (vary in # anchors).
# (3) Ultimately, zeb_integrated_50 is used for tissue mapping.

# Other Contents:
# DEGs from Wagner Supplement Table 2 for all tissues.
# Batch correction plots, summary stats.

# Notes:
# Unused markers (Zfin, Andrea) at bottom.
# zeb_integrated_* (10, 30 , 70) have been removed from .Renv.

# Segways into: zeb_degs_heatmaps.Rmd.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r include=TRUE}
library(Seurat)
library(dplyr)
library(ggplot2)
library(purrr)
library(Hmisc)
library(pvclust)
```

############  KLEIN ZEBRAFISH ######### 

# Extract individual Raw UMI-filtered counts (csv). Example below. 
# https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM3067195
# GSM3067195_24hpf.csv.gz

# [LOC100000009 x DEW050_GTCTCTCT_AGGAGCTT]
# 30677 x *
```{r include=TRUE}
zeb_24hpf_dir <- "/home/pprakriti/princeton_google_drive/Levine Lab/Orthology-Maps/Raw_Data/Klein_2018/GSM3067195_24hpf.csv" 

# 7970
zeb_24hpf_df <- read.table(zeb_24hpf_dir,
                          sep = ",",
                          header = TRUE,
                          stringsAsFactors = FALSE)

# Clean up the dataframe
rownames(zeb_24hpf_df) <- zeb_24hpf_df$Row
zeb_24hpf_df$Row <- NULL
```

# Helper Function to make above raw matrix data frames
```{r include=TRUE}
make_zeb_df <- function(zeb_dir) {
  zeb_df <- read.table(zeb_dir,
                       sep = ",",
                       header = TRUE,
                       stringsAsFactors = FALSE)
  
  rownames(zeb_df) <- zeb_df$Row
  zeb_df$Row <- NULL
  
  return(zeb_df)
}
```


"""
Make all raw matrix data frames
"""

# T = 34750 (exactly what is reported in Fig S2B)
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM3067196,  
```{r include=TRUE}
zeb_t1_dir <- "/home/pprakriti/princeton_google_drive/Levine Lab/Orthology-Maps/Raw_Data/Klein_2018/GSM3067196_TracerSeq1.csv"
# 5752
zeb_t1_24hpf_df <- make_zeb_df(zeb_t1_dir)
```

https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM3067197,
```{r include=TRUE}
zeb_t2_dir <- "/home/pprakriti/princeton_google_drive/Levine Lab/Orthology-Maps/Raw_Data/Klein_2018/GSM3067197_TracerSeq2.csv"
# 8874
zeb_t2_24hpf_df <- make_zeb_df(zeb_t2_dir)
```

https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM3067198,
```{r include=TRUE}
zeb_t3_dir <- "/home/pprakriti/princeton_google_drive/Levine Lab/Orthology-Maps/Raw_Data/Klein_2018/GSM3067198_TracerSeq3.csv"
# 2861
zeb_t3_24hpf_df <- make_zeb_df(zeb_t3_dir)
```

https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM3067199,
```{r include=TRUE}
zeb_t4_dir <- "/home/pprakriti/princeton_google_drive/Levine Lab/Orthology-Maps/Raw_Data/Klein_2018/GSM3067199_TracerSeq4.csv"
# 6495
zeb_t4_24hpf_df <- make_zeb_df(zeb_t4_dir)
```

https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM3067200
```{r include=TRUE}
zeb_t5_dir <- "/home/pprakriti/princeton_google_drive/Levine Lab/Orthology-Maps/Raw_Data/Klein_2018/GSM3067200_TracerSeq5.csv"
# 2844
zeb_t5_24hpf_df <- make_zeb_df(zeb_t5_dir)
```


"""
Let's pool all the data together and batch correct raw matrix values
"""

# https://satijalab.org/seurat/v3.1/integration.html
# (1) Perform standard pre-processing steps on each Seurat object.
```{r include=TRUE}
make_zeb_seurat <- function(zeb_df, min_cells, min_features, n_features){
  zeb_seurat <- CreateSeuratObject(counts = zeb_df,
                                   min.cells = min_cells, 
                                   min.features = min_features)
  
  zeb_seurat <- NormalizeData(object = zeb_seurat)
  
  zeb_seurat <- FindVariableFeatures(zeb_seurat, 
                                     selection.method = "vst",
                                     nfeatures = n_features)
  return(zeb_seurat)
  }
```

# (2) Make the individual Seurat Objects
```{r include=TRUE}
zeb_min_cells <- 0
zeb_min_features <- 200
zeb_n_features <- 2000

zeb_24hpf_seurat <- make_zeb_seurat(zeb_24hpf_df,
                                    zeb_min_cells,
                                    zeb_min_features,
                                    zeb_n_features)

zeb_t1_24hpf_seurat <- make_zeb_seurat(zeb_t1_24hpf_df,
                                    zeb_min_cells,
                                    zeb_min_features,
                                    zeb_n_features)

zeb_t2_24hpf_seurat <- make_zeb_seurat(zeb_t2_24hpf_df,
                                    zeb_min_cells,
                                    zeb_min_features,
                                    zeb_n_features)

zeb_t3_24hpf_seurat <- make_zeb_seurat(zeb_t3_24hpf_df,
                                    zeb_min_cells,
                                    zeb_min_features,
                                    zeb_n_features)

zeb_t4_24hpf_seurat <- make_zeb_seurat(zeb_t4_24hpf_df,
                                    zeb_min_cells,
                                    zeb_min_features,
                                    zeb_n_features)

zeb_t5_24hpf_seurat <- make_zeb_seurat(zeb_t5_24hpf_df,
                                    zeb_min_cells,
                                    zeb_min_features,
                                    zeb_n_features)
```


"""
Make list of above Seurat Objects
"""

```{r include=TRUE}
zeb_seurat_list <- list(zeb_24hpf_seurat, zeb_t1_24hpf_seurat, zeb_t2_24hpf_seurat, zeb_t3_24hpf_seurat, zeb_t4_24hpf_seurat, zeb_t5_24hpf_seurat)
```


"""
Integration Routine- First pass on 30 anchors
"""

```{r include=TRUE}
# ~30 mins
zeb_anchors <- FindIntegrationAnchors(object.list = zeb_seurat_list,
                                      dims = 1:30)
```
```{r include=TRUE}
zeb_integrated <- IntegrateData(anchorset = zeb_anchors, dims = 1:30)
```

```{r include=TRUE}
DefaultAssay(zeb_integrated) <- "integrated"

# Run the standard workflow for visualization and clustering
zeb_integrated <- ScaleData(zeb_integrated, verbose = FALSE)
zeb_integrated <- RunPCA(zeb_integrated, npcs = 70, verbose = FALSE)
zeb_integrated <- RunTSNE(zeb_integrated, dims = 1:70)

```

```{r include=TRUE}
zeb_integrated <- FindNeighbors(zeb_integrated, dims = 1:70)
zeb_integrated <- FindClusters(zeb_integrated, resolution = 2)
```


"""
Turn above routine into a function
"""

# Helper Function
```{r include=TRUE}
make_integrated_seurat <- function(object_list, dims, npcs){
  # Args:
    # object_list: list of Seurat objects. e.g. zeb_seurat_list.
    # dims: input dimensions for FindIntegrationAnchors and IntegrateData. 
    # npcs: # pcs for PCA and TSNE.
  
  # Routine: Do integration. Scale, PCA, TSNE.
  
  # Returns:
    # integrated_seurat: output of above routine.
  
  zeb_anchors <- FindIntegrationAnchors(object.list = object_list,
                                      dims = 1:dims)
  zeb_integrated <- IntegrateData(anchorset = zeb_anchors, dims = 1:dims)
  
  DefaultAssay(zeb_integrated) <- "integrated"
  
  zeb_integrated <- ScaleData(zeb_integrated, verbose = FALSE)
  zeb_integrated <- RunPCA(zeb_integrated, npcs = npcs, verbose = FALSE)
  zeb_integrated <- RunTSNE(zeb_integrated, dims = 1:npcs)
  
  return(zeb_integrated)
}
```


"""
Make Various Integrated Objects
"""

# dims=10, npcs=70 - Done
```{r include=TRUE}
zeb_integrated_10 <- make_integrated_seurat(zeb_seurat_list, 10, 70)
```

# dims=50, npcs=70 - Done
```{r include=TRUE}
zeb_integrated_50 <- make_integrated_seurat(zeb_seurat_list, 50, 70)
```

# dims=70, npcs=70 - Done
```{r include=TRUE}
zeb_integrated_70 <- make_integrated_seurat(zeb_seurat_list, 70, 70)
```

"""
Cluster Them
"""

# 50 - Done
```{r include=TRUE}
zeb_integrated_50 <- FindNeighbors(zeb_integrated_50, dims = 1:70)
zeb_integrated_50 <- FindClusters(zeb_integrated_50, resolution = 2)
```

# 70 - Done
```{r include=TRUE}
zeb_integrated_70 <- FindNeighbors(zeb_integrated_70, dims = 1:70)
zeb_integrated_70 <- FindClusters(zeb_integrated_70, resolution = 2)
```

"""
DEGs to Use: Supp Table 2
"""

```{r include=TRUE}
# noto- DONE
noto_50_degs <- c("ntd5", "aplp2", "emilin3a", "pmp22b", "loxl5b", "ta", "si:dkey-12l12.1", "plod1a", "calua", "fstl1b", "fn1a", "LOC100333762", "col2a1a", "serpinh1b", "fxyd1", "LOC100149028", "fstl1a", "col11a1a", "cmn", "col2a1b", "emilin3a", "si:dkey-12l12.1", "loxl5b", "ta", "LOC100333762", "LOC100149028", "ntd5", "fxyd1", "cmn", "pmp22b", "fn1a", "plod1a", "serpinh1b", "calua", "aplp2", "col11a1a", "fstl1a", "col2a1a", "fstl1b", "col2a1b")


# Hearts (3) # - DONE
heart_51_degs <- c("si:dkey-261h17.1", "LOC100331639", "tpm4a", "krt18", "krt8", "pcolcea", "flna", "ftr82", "fn1a", "si:dkey-207j16.2", "rbpms2b", "krt94", "podxl", "myl9b", "LOC563082", "hand2", "colec10", "sfrp5", "fabp11a"	, "si:dkey-261h17.1", "pcolcea", "tmem88b", "krt18", "ftr82", "LOC100331639", "fn1a", "si:dkey-207j16.2", "krt8", "flna", "tpm4a", "rbpms2b", "krt94", "LOC563082", "myl9b", "hand2", "podxl", "colec10", "sfrp5", "fabp11a")

heart_mature_52_degs <- c("tpm4a", "fhl2a", "tnnt2a", "myl7", "tnnc1a")

heart_hox9da_56_degs <- c("si:dkey-261h17.1", "pmp22a", "prrx1a", "reck", "prdm1a", "aldh1a2")



# Muscles (3) # - DONE
mus_myl10_53_degs <- c("tpm3", "tpm2", "hm:zewp0073", "actc1a", "myl13")

mus_myl1_12_degs <- c("mylz3", "actc1a", "hm:zewp0073", "myl1", "mylpfa", "tpma", "tnnc2", "acta1b", "ckmb", "ckma", "actc1b")

myotome_27_degs <- c("hsp90aa1.1", "hm:zewp0073", "actc1a", "klhl41b", "myog", "pabpc4", "tnni2b.1", "myl1", "hspb1")



# Pancreas - DONE
panc_prim_4_degs <- c("slc1a4", "pax6b", "anxa4", "slc43a2a", "scg3", "isl1", "tspan7b", "rltgr", "scgn", "nucb2b", "neurod1", "LOC565781", "si:dkey-153k10.9")



# pharyngeal stuff (5) # - DONE
# same first 10
pharyn_pouch_28_degs <- c("epcam", "pax1b", "si:dkey-261h17.1", "krt18", "krt8", "egfl6", "etv4", "prdm1a", "cd9b", "nanos1", "alcama", "cfl1l", "cldn7b", "cxcl12b", "tbx1", "foxa2", "hoxb1b", "nkx2.3", "si:ch211-67e16.11", "cxcl12a", "pax1b")

pharyn_arch_tbx1_31_degs <- c("epcam", "pax1b", "si:dkey-261h17.1", "krt18", "krt8", "egfl6", "etv4", "prdm1a", "cd9b", "six1b", "npm1a", "cyp26c1", "her6", "eya1", "nrp2a", "pdgfab", "nrp2b", "cxcl12a", "igfbp5b", "sfrp2")

pharyn_arch_cd248b_35_degs <- c("pmp22a", "cd248b", "col1a1a", "col1a2", "tgfbi", "fibina", "reck", "tpm4a", "cthrc1a")

pharyn_arch_ndnf_55_degs <- c("pitx3", "si:dkey-261h17.1", "ndnf", "eya2", "pmp22a", "foxc1a", "fgfr4", "foxc1b", "tmem88b")

pharyn_arch_lbx1a_58_degs <- c("thbs4b", "vwde", "six1b", "postnb", "eya1", "sept9a", "lbx1a", "rbp5", "met")



# Epi's (12) # - DONE
epi_prr15la_3_degs <- c("icn", "pvalb8", "cldnh", "prr15la", "cldnb", "zgc:92380", "icn2", "epcam", "LOC100006250") 

epi_grhl3_22_degs <- c("si:dkey-87o1.2", "wu:fb12c09", "cldne", "icn", "ponzr5", "cldnb", "spint2", "icn2", "s100u")

epi_olf_23_degs <- c("cldnh", "epcam", "f11r.1", "pvalb8", "cldnb", "cldn7b", "capsla", "insm1b", "tspan2a")

epi_ant_29_degs <- c("cfl1l", "epcam", "col1a2", "krt8", "cldn1", "cldni", "postnb", "padi2", "pfn1")

epi_atp1a1a_2_41_degs <- c("LOC569335", "atp1b1b", "ndrg1a", "cldnh", "atp1a1a.2", "icn", "ndrg3a", "cldne", "spint2")

epi_muc5ac_57_degs <- c("agr2", "s100u", "icn", "foxa3", "krt97", "icn2", "cldnb", "zgc:92380", "zgc:100864")

epi_s100a11_66_degs <- c("atp1a1a.1", "s100a11", "trpv6", "ponzr5", "LOC569335", "atp1b1b", "cldnh", "mafbb", "pvalb8")

epi_and1_68_degs <- c("egfl6", "apoeb", "cfl1l", "pfn1", "cldni", "col1a1a", "krt97", "cldn1", "s100a10b")

epi_acbd7_69_degs <- c("llgl2", "zgc:77517", "cldnb", "icn", "cldne", "acbd7", "cldnh", "si:ch211-195b11.3", "oclna")

epi_rbp4_70_degs <- c("pfn1", "LOC100334363", "cldni", "s100a10b", "cotl1", "cfl1l", "si:dkey-152p16.6", "col1a1b", "krt4")

epi_col7a1l_71_degs <- c("pfn1", "cfl1l", "tmsb4x",	"krt4", "cyt1", "cyt1l", "si:dkey-152p16.6", "cldni", "cldn1")

epi_kera_72_degs <- c("cldni", "kera", "col1a2", "col1a1b", "col11a1a", "LOC100334363", "cfl1l", "col1a1a", "pfn1")

# Differentiating Neurons #
diff_neurons_hmx_1_degs <- c("celf3a", "sox1a", "LOC100536821", "rtn1a", "sox1b", "LOC103911504", "hmx3", "hmx3a", "stmn1b")

diff_neurons_dlx_2_degs <- c("dlx1a", "dlx5a", "dlx2a", "celf3a", "elavl3", "sp9", "LOC100536821", "rtn1a", "dlx2b")

diff_neurons_phox2a_9_degs <- c("phox2a", "phox2bb", "elavl3", "vim", "nkx6.1", "onecut1", "insm1a", "dlb", "nhlh2")

diff_neurons_11_degs <- c("elavl3", "onecut1", "tubb5", "LOC100536821", "myt1a", "ebf2", "rtn1a", "nhlh2", "dlb")

diff_neurons_eomesa_34_degs <- c("tbr1b", "eomesa", "celf3a", "elavl3", "stmn1b", "rtn1a", "r3hdm1", "celf2", "bcl11ba")

diff_neurons_sst1.1_40_degs <- c("scg3	im:7138475", "tal2", "gad2", "rhpn2", "gata3", "nppc", "myo3b", "gad1b")

diff_neurons_rohon_beard_43_degs <- c("tuba8l3", "neurod1", "six1b", "zgc:158291", "cdh6", "pcbp4", "ebf3", "onecut1", "onecutl")

diff_neurons_rohon_beard_43_degs



# Neural #
neural_midbrain_13_degs <- c("eng2b", "sox19a", "otx2", "wu:fb25b09", "otx1a", "LOC103911157", "efna2a", "sox3", "otx1b")

neural_v_hindbrain_7_degs <- c("sp8b", "lrrn1", "sox3", "gfap", "plp1a", "LOC100334443", "pax6a", "si:ch73-21g5.7", "sox19a")

neural_dorsal_spinal_cord_14_degs <- c("zslc1a3a", "si:ch211-251b21.1", "mdka", "mdkb", "prss35", "gfap", "hoxc3a", "hoxb9a", "hoxa4a")

neural_dienceph_15_degs <- c("id3", "nkx2.4a", "prdx1", "nkx2.4b", "fezf2", "shha", "rx3", "nkx2.1", "sox1b")

neural_d_hindbrain_24_degs <- c("casz1", "mdkb", "zic1", "olig3", "zic2b", "zic2a", "epha2a", "fzd10", "cdon")

neural_floorplate_37_degs <- c("hapln1a", "slit2", "slit1b", "spon1b", "sparc", "col2a1a", "sulf1", "npr3", "ntn1b")

neural_floorplate_post_39_degs <- c("LOC103910062", "slc1a3a", "sulf1", "col4a5", "vim", "col11a1a", "foxj1a", "nkx2.2b", "col14a1a")

neural_hind_roofplate_42_degs <- c("wls", "cdon", "mdka", "cdh6", "atp1b3a", "rspo3", "igfbp7", "zic2a", "bmp5")

neural_telenceph_45_degs <- c("lrig1", "foxg1a", "LOC100538174", "fezf2", "cyp26c1", "mdkb", "sox1b", "sox5", "LOC100536415")

neural_mid_v_nkx6_2_46_degs <- c("ntn1a", "nkx6.2", "foxa1", "lrrn1", "ntn1b", "wu:fb25b09", "foxa", "otx1a", "sulf2b")

neural_hind_gsx1_47_degs <- c("mdka", "lbx1b", "gsx1", "mdkb", "LOC100150849", "irx3a", "si:ch211-137a8.4", "sox19a", "plp1a")

neural_post_v_nkx6_2_48_degs <- c("ntn1a", "sulf2b", "nkx6.2", "nkx2.2b", "phox2bb", "nkx6.1", "nkx2.9", "ntn1b", "nkx2.2a")

neural_dienceph_post_54_degs <- c("shha", "pitx3", "foxa1", "fgfr3", "ntn1b", "barhl1a", "foxa2", "pitx2", "lmx1ba")



# Neural Crests # - DONE
nc_16_degs <- c("crestin", "sox10", "tspan36", "ctsba", "slc2a15b", "cax1", "slc45a2", "slc2a11b", "pah")

nc_xanth_59_degs <- c("impdh1b", "paics", "gch2", "gmps", "prdx1", "wu:fc46h12", "slc2a11b", "akr1b1", "LOC100536795")

nc_mel_60_degs <- c("atp6v0ca", "atp6v1g1", "gstp1", "tspan36", "LOC103911087", "slc45a2", "si:ch211-102c2.5", "slc37a2",	"tyrp1b")

nc_irid_61_degs <- c("crestin", "gmps", "apoda.1", "si:ch211-243a20.3", "defbl1", "si:dkey-197i20.6", "pnp4a", "impdh1b", "sox10")

nc_grem2_62_degs <- c("pmp22a", "twist1a", "si:dkey-261h17.1", "ednrab", "grem2b", "her6", "grem2", "dlx2a", "npm1a") 

nc_mcamb_63_degs <- c("ednrab", "sox10", "crestin", "foxd3", "tuba8l3", "erbb3b", "pmp22a", "tfap2a", "LOC103910711")



# Optic Stuff #
lens_36_degs <- c("fabp11a", "crybb1", "tgfb3", "vcana", "hmx4", "pitx3", "lim2.1")

optic_cup_17_degs <- c("rx1", "six3b", "ncl", "pcna", "cad", "nap1l1", "fabp7a", "tktb", "hspe1")

rpe_44_degs <- c("fabp11a", "pmela", "dct", "LOC103911087", "cx43", "pmelb", "tspan36", "gstp1", "bhlhe40")

otic_plac_25_degs <- c("stm", "si:dkey-222f2.1", "oc90", "sox10", "si:ch211-152c2.3", "cldna", "si:ch211-80h18.1", "cldnb", "ptbp1b")

otic_plac_25_degs



```

```{r include=TRUE}
# Immune+Blood # 
iono_5_degs <- c("atp6v0ca", "atp6v1ba", "atp6v1g1", "atp6v1aa", "LOC569335", "atp6ap1b", "atp6v0a1a", "foxi3a", "ca15a")

leuko_6_degs <- c("cd63", "lyz", "lta4h", "ncf1", "papss2b", "coro1a", "srgn", "si:ch1073-429i10.1", "mpx")

eryth_8_degs <- c("si:ch211-5k11.12", "hbae3", "hbbe1.1", "hbbe3", "MGC173646", "LOC100334599", "si:ch211-5k11.2", "hbae1", "slc4a1a")

macro_21_degs <- c("ctsba", "cd63", "LOC100537803", "lcp1", "cotl1", "coro1a", "arpc1b", "ctssb.1", "mfap4")




# endo 
endo_26_degs <- c("slc1a4", "pax6b", "anxa4", "slc43a2a", "scg3", "isl1", "tspan7b", "rltgr", "scgn", "nucb2b", "neurod1", "LOC565781", "si:dkey-153k10.9")



# Endothelial #
endoth_32_degs <- c("clec14a", "yrk", "krt18", "fli1a", "etv2", "tmem88a", "kdrl", "si:ch211-156j16.1", "si:dkey-261h17.1")

endoth_post_38_degs <- c("ctsla", "stab2", "dab2", "mrc1a", "glula", "snx5", "si:ch211-156j16.1", "clec14a", "flt4")

# Tailbud #
tail_spinal_cord_19_degs <- c("hoxb9a", "mdka", "ldlrad2", "slc1a3a", "hoxc3a", "gfap", "pkd1b", "LOC100334443", "LOC100001360")

tail_psm_30_degs <- c("apoc1l", "wu:fb77a09", "LOC100537277", "apoeb", "her1", "ntd5", "fn1b", "tuba8l2", "hoxc13b")



```

```{r include=TRUE}

# Misc # 

meso_emp2_10_degs  <- c("col5a2a", "vwde", "emp2", "col5a1", "thbs4b", "col1a2", "col1a1b", "angptl7", "tgfbi")

peri_18_degs <- c("cyt1l", "krt4", "cyt1", "wu:fb18f06", "anxa1c", "cst14b.1", "krt17", "si:ch211-195b11.3", "zgc:193505")

procto_65_degs <- c("atp1b1a", "cldnc", "mal", "cldnh", "elf3", "spint2", "cldnb", "rbm47", "ponzr1")

pro_duct_64_degs <- c("atp1b1a", "aldob", "slc4a4a", "slc4a2a", "LOC101883320", "rbm47", "atp1a1a.4", "ndrg1a", "si:dkeyp-27c8.2")

lat_line_krt15_67_degs <- c("krt15", "eya1", "cldnb", "hmx3", "hmx3a", "epcam", "gsnb", "si:dkey-222f2.1", "igsf9")

germ_49_degs <- c("nanos3", "dnd1", "ddx4", "zgc:56699", "tdrd7a", "ca15b", "LOC100330598", "fbxo18", "psap")

pec_fin_bud_20_degs <- c("pcolcea", "hand2", "cfd", "pitx1", "si:dkey-261h17.1", "ftr82", "col5a1", "pmp22a", "tmem88b")

pec_fin_bud_20_degs
```


"""
Let us identify cell types/tissues using zeb_integrated_50
"""

```{r include=TRUE}
zeb_features <- pec_fin_bud_20_degs[1:9]

FeaturePlot(object = zeb_integrated_50,
            features = zeb_features,
            reduction = "tsne",
            label = FALSE)
```


"""
Sanity Checks Below
"""

"""
Check if Batch Effect Correction was successful
"""
# Below is to see if any cluster is enriched with cells from a single batch.
# Seems like things are well overlapped.
# https://github.com/satijalab/seurat/issues/1242
```{r include=TRUE}
zeb_integrated_seurat <- zeb_integrated_50

DimPlot(zeb_integrated_seurat, 
        reduction = "tsne",
        label = TRUE,
        group.by = "orig.ident")
```

# For 50 anchors, there are 76 clusters. 
# 29 libraries/batches for all. 
```{r include=TRUE}
test_table_50 <- table(zeb_integrated_50$orig.ident, zeb_integrated_50$integrated_snn_res.2)
test_df_50 <- as.data.frame(test_table_50)

test_table_70 <- table(zeb_integrated_70$orig.ident, zeb_integrated_70$integrated_snn_res.2)
test_df_70 <- as.data.frame(test_table_70)
```

# The histogram corresponding to an ideal batch correction would show a uniform distribution of percent contribution of ech batch to a cluster. Below is my hand-curated attempt to visualize these distributions for 50 and 70 anchors. I cannot say that 70 is necessarily better than 50, so I will continue my analyses with zeb_integrated_50.
```{r include=TRUE}
# Want to see the perrcent contribution of each batch to each cluster
var_num <- 5

test_cluster_50 <- test_df_50[test_df_50$Var2 == var_num, ]
test_cluster_50_dp <- test_cluster_50 %>% mutate(
  av_contribution = Freq/sum(Freq)*100)

test_hist_50 <- hist(test_cluster_50_dp$av_contribution)
test_hist_50

test_cluster_70 <- test_df_70[test_df_70$Var2 == var_num, ]
test_cluster_70_dp <- test_cluster_70 %>% mutate(
  av_contribution = Freq/sum(Freq)*100)

test_hist_70 <- hist(test_cluster_70_dp$av_contribution)
test_hist_70
```


"""
End
"""


"""
Other Markers (Unused)
"""

# ZFIN CURATED GENES # 
```{r include=TRUE}
# from zfin https://zfin.org/ZFA:0000114
zfin_heart_genes <- c("myl7", "nppa", "bmp4", "hspb1", "myh6")

zfin_noto_genes <- c("ta", "shha", "col2a1a", "lft1", "col9a2", "angptl6")
```

# ALL ANDREA MARKERS #
```{r include=TRUE}
# Observations for:
# npcs/TSNE/FindNeighbors = 70; resolution = 4

# heart primordium- 11
andrea_heart_prim_markers <- c("hand2", "gata5", "gata6")

# noto- no idea. High expression in 19 (etv4). (Maybe 33 using klein_degs)
andrea_noto_markers <- c("ta", "ndt5", "col2a1a", "clo8a1a", "p4ha1b", "etv4", "col5a3a", "prx")

# No expression.
andrea_pgc_markers <- c("ddx4", "nanos3", "dnd1", "piwi2", "h1m")

# I think kind of confidently 39 
andrea_hema_markers <- c("tal1", "lmo2", "spi1b")

# I think 17?
andrea_tailbud_markers <- c("dlx5a", "hexb", "mgat1b", "ntla", "cyp26a1", "tll1", "eve1", "bmp2a", "tagln3b", "szl", "abcc6a", "CABZ01079764.1", "bambib", "dusp4")

# Completely unclear (no localized expression) 
andrea_cef_meso_markers <- c("fsta", "foxf2a")

# Maybe 19 (low expression)
andrea_pharyngeal_markers <- c("nkx2.7", "irx7")

# No expression
andrea_hepatocyte_markers <- c("fabp10a", "cp", "hmgcra", "fads2", "msmo1")

# 35 (high confidence)
andrea_pancreas_markers <- c("cdx4")

# 38 v. strong, 17 too 
andrea_hatching_gland_markers <- c("ctslb", "he1a", "svopl", "tram1", "fam46ba", "cd63", "calrl", "tmprss9", "he1b", "rcv1")

# Unclear. btg2 everywhere.
andrea_seg_plate_markers <- c("fgf17", "tjp2b", "kazald2", "btg2", "si:ch211-196c10.11", "si:dkey-14o18.1", "par1", "dmrt2a", "meox1", "si:ch211-105d18.8", "si:rp71-36n21.1", "crabp2b", "CU633857.1")

# No expression anywhere
andrea_heart_general_markers <- c("atp2a2a", "cryba1l1", "fhl2a", "myl7", "tnni1b", "tnnt2a", "vmhc", "cmlc1", "tbx20", "cyp26c1", "crygn2", "mafba", "tnnc1a", "podxl")

# Pax2a good expression in 1.
andrea_pronephor_markers <- c("foxj1a", "pax2a")

# Everywhere, unclear
andrea_somites_markers <- c("mef2ca", "ube2h", "fam107b", "xirp2a", "hapln1a", "ctgfa", "si:dkey-117n7.3", "kcnn3", "s1pr5a", "junba", "lama2", "si:ch211-207e19.11", "ckma", "tnnc1b")

# 24/42
andrea_neural_crest_markers <- c("foxd3", "sox9b", "sox10")

# Unclear, some expression in 0- not in 1 general area though
andrea_floor_plate_markers <- c("shha", "shhb", "foxj1a", "foxa2")

# v. strong + consistent expression in 13->33+37+46
andrea_spinal_cord_markers <- c("elavl3", "neurod1", "neurod4", "neurog4", "neurog1")

# v. convincing expression in 3, 27, 43, 29
andrea_epi_markers <- c("foxi3a", "foxi3b", "myb", "gcm2", "gata2a", "tbx2b", "cyp2k16")

# Could buy that it's 25
andrea_np_border_markers <- c("crabp2a", "dlx3b")

# Strong 1 (eng2a, pax2a)
# Nice non-overlapping clusters 32+34 (olig3)
andrea_brain_markers <- c("eng2a", "her5", "pax2a", "emx3", "rx3", "pax6b", "olig3")

# 27, 43, 3, 29 + strong 17 (same as epi?!)
andrea_forebrain_markers <- c("gstm", "uck2b", "relnl", "aldh1a3", "barhl1a", "cx43", "arl3l1", "otx2", "foxe3", "foxd1", "nkx2.1a", "zgc:171531", "rx2", "vsx2")

# No expression- completely unclear 
andrea_newest_somite_markers <- c("ripply1", "mespba", "ripply2", "mespab", "tsc22d2")

# No expression, some in 0
andrea_early_endo_markers <- c("foxa1", "foxa2", "sox17", "sox32", "mixl1", "prdx5")

# Present in "overlapping" clusters (0, brain cluster, 20+23+28)
andrea_general_meso_markers <- c("myl10", "myod1", "myog", "myf5", "meox1", "ripply1", "aldh1a2", "ripply2", "mespba", "tbx6", "tbx6l")
```

# These are recommended markers from Zfin
```{r include=TRUE}
zfin_rec_heart <- c("casz1", "comp", "fgf8a", "gata5", "gata6")
zfin_rec_noto <- c("angptl6", "chad", "col9a2", "col27a1a", "cxcr4b")
```


"""
Troubleshooting, Supp pg 3 (Did not do)
"""
# Highly variable genes -> Fano factors, then rank all genes by above-Poisson noise statistic
## Take top 2000 genes
## Prune list for highly correlated genes and housekeeping/cell-cycle genes.
## 1) If corr(transcript, any other transcript) > 0.2
## 2) Ref list for cell cycle + housekeeping + anything > 0.4 corr 
## 3) Another round of > 0.4 corr pruning
# PCA (skip)
## # PCs ~ compare eigenvalue distribution of cell PCs to that of randomized data   

# Clustering
## Supp pg 11- get a sense of numbers/cluster neighbors 
## Local density clustering

## Important points: (skip)
### Removed 1 cluster with only housekeeping genes
### If 2 clusters didn't have DEGs -> merge

# TSNE: Perplexity=30, 1000-5000 iterations! 
