---
title: "get_gene_set_troubleshooting"
output: html_document
---
# Follows from: "get_pairwise_zeb_ciona_degs.Rmd".
# Note: This exploratory script exists because only 6 1-1 DEG Orthologs were identified (implementation present at end).

# Works in parallel with: None.

# General Description: Script implements various methods to get gene set
# to be used for downstream cross-species cell type comparisons. 

# VVIMP: This Script contains entire Tosches workflow = 
# Gene Specificity Matrix, Spearman Correlations. 
# However, Tosches worflow is run on 25 1-1 orthologs, NOT 1-1 DEGs!

# Uses env_get_pairwise_zeb_ciona_degs.RData

# Module v2: Wrong Implementation. Used "get_geneset_v2.py" (at end of Script)
# Important Observations: (at end of Script)
#   (1) Only Var Genes are used for DEG calculations can be visualized. 
#   (2) zeb_integrated_50 and clean_chen_larva_integrated_20 have 2000.

# Module v3 implemented 7/2/20 using 25 genes.
# We do not need to use 2000 as the upper bound -> 
# get gene expression values of all 1-1 orthologs.  

# Segways into: "get_geneset_v4.Rmd" 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r include=FALSE}
library(Seurat)
library(dplyr)
library(ggplot2)
```

""""""""""""""""""
Module v3: Use all expressed 1-1 Orthologs using 2000 VarFeatures (as the upper bound)
""""""""""""""""""

"""
Get KHIDs and Zeb Genes 
"""
## Vary ##
```{r include=TRUE}
# Older Implemenation

khid_list_dir = "/home/pprakriti/princeton_google_drive/Levine Lab/Orthology-Maps/get_geneset_attempts/v3/khid_list.csv" 

zeb_genes_dir = "/home/pprakriti/princeton_google_drive/Levine Lab/Orthology-Maps/get_geneset_attempts/v3/zeb_genes.csv" 

khid_list_df <- read.table(khid_list_dir,
                           sep = ",",
                           header = FALSE,
                           stringsAsFactors = FALSE)

zeb_genes_df <- read.table(zeb_genes_dir,
                           sep = ",",
                           header = FALSE,
                           stringsAsFactors = FALSE)

khid_list = khid_list_df$V1
zeb_genes = zeb_genes_df$V1
```

```{r include=TRUE}
# Implementation Used

one_one_dir = "/home/pprakriti/princeton_google_drive/Levine Lab/Orthology-Maps/get_geneset_attempts/v3/one_to_one_khid_zeb_gene_dict.csv" 

one_one_df <- read.table(one_one_dir,
                         sep = ",",
                         header = FALSE,
                         stringsAsFactors = FALSE)

khid_list = one_one_df$V1
zeb_genes = one_one_df$V2

```

"""
Which of these are expressed in Ciona and Zeb?
"""
## Can Vary Seurat Object ##
```{r include=TRUE}
# 141
ciona_vars <- VariableFeatures(clean_chen_larva_integrated_20)
ciona_ortho_vars <- intersect(ciona_vars, khid_list)

# 81
zeb_vars <- VariableFeatures(zeb_integrated_50)
zeb_ortho_vars <- intersect(zeb_vars, zeb_genes)
```

"""
Get the 81 KHIDs that correspond to the expressed Zeb genes.
"""
```{r include=TRUE}
test <- filter(one_one_df, V2 %in% zeb_ortho_vars)
khid_zeb_ortho_vars <- test$V1
```

"""
Further munging
"""
```{r include=TRUE}
# These are zeb orthologs that are expressed in ciona - 25 
final_khids <- intersect(khid_zeb_ortho_vars, ciona_vars)
```

```{r include=TRUE}
test_2 <- filter(one_one_df, V1 %in% final_khids)
final_zeb_genes <- test_2$V2
```

"""
Let's visualize the profiles using this gene set.
"""
```{r include=TRUE}
# slot = "data"
DoHeatmap(object = clean_chen_larva_integrated_20,
          features = final_khids,
          cells = ciona_larva_noto)
```

```{r include=TRUE}
# slot = "data"
DoHeatmap(object = zeb_integrated_50,
          features = final_zeb_genes,
          cells = noto)
```
```{r include=TRUE}
# slot = "data"
DoHeatmap(object = clean_chen_larva_integrated_20,
          features = final_khids)
```

"""""""""""""""""""""""""""""""" 
Let us implement Tosches.
In Module v3, we have 25 genes. 
""""""""""""""""""""""""""""""""

""""""""""""""""""""""""""""""""
 START HELPER FUNCTIONS
""""""""""""""""""""""""""""""""

"""
Helper Functions to make Average Expression Matrix
"""
```{r include=TRUE}
get_all_total_size <- function(cluster_sizes) {
  # Given a vector of cluster sizes, get total size i.e. total number
  # of cells in ALL CTs.
  # Note: "all" prepends "total_size" because var "total_size" is present  
  #       "make_columns_2 function" which counts number of cells in ONE CT.
  # Args:
  #   cluster_sizes: Output of table(Idents(test_obj)).
  #
  # Returns:
  #   total_size: Sum of all values in cluster_sizes.
  all_total_size <- sum(cluster_sizes)
  return(all_total_size)
}
```

```{r include=TRUE}
make_av_exp_matrix <- function(integrated_seurat_object, txnptomic_profile) {
  # Given  an Integrated Seurat Object and "transcriptomic profile", i.e. genes whose expression
  # we are interested in, we make an "Average Expression Matrix" (av_exp_matrix).
  #
  # Args:
  #   integrated_seurat_object: Self-explanatory. 
  #     e.g. clean_chen_larva_integrated_20.
  #   txnptomic_profile: Vector of gene names.
  #     e.g. khid_final (25 1-1 orthologs with 2000 VarFeatures)
  #
  # Returns:
  #   av_exp_matrix: Rows: genes in txnptomic profile.
  #                  Columns: average expression of genes in given cluster.
  #                  Note: Colnames have "cluster flags".
  av_exp_matrix <- AverageExpression(integrated_seurat_object,
                                     features = txnptomic_profile)$RNA
  colnames(av_exp_matrix) <- paste("cluster", colnames(av_exp_matrix), sep = "_")
 
  return(av_exp_matrix) 
}
```

```{r include=TRUE}
make_cluster_flags <- function(cluster_nums) {
  cluster_flags <- c()

  for (cluster_num in cluster_nums) {
    cluster_flag <- paste("cluster_", cluster_num, sep = "")
    cluster_flags <- c(cluster_flags, cluster_flag)
  }
  return(cluster_flags)
}
```

```{r include=TRUE}
make_columns_2 <- function(columns_1, cluster_nums, cluster_sizes) {
  # Notes:
  # cluster_nums: cluster numbers that correspond to a CT.
  # cluster_size: total cells in a CT. 
  columns_2 <- columns_1
  # all cells in a CT.
  total_size <- 0
  
  for (i in 1:length(cluster_nums)) {
    # cluster under inspection 
    cluster_num <- cluster_nums[i]
    # this will be the index in ciona_cluster_sizes table that will give you the number of cells in a cluster.
    cluster_size_num <- cluster_nums[i]+1
    cluster_size <- cluster_sizes[cluster_size_num]
    
    # multiply size with everything in its column to get total expression of each gene in that cluster. 
    columns_2[, i] <- columns_2[, i]*cluster_size 
    
    # get total number of cells in a cell type. 
    total_size <- total_size+cluster_size
  }
  
  to_return <- list(columns_2, total_size)
  return(to_return)
}
```

```{r include=TRUE}
make_columns_3 <- function(columns_2, total_size) {
  columns_3 <- columns_2 %>% mutate(total_exp = rowSums(.)) %>% mutate(av_exp = total_exp/total_size)
  rownames(columns_3) <- rownames(columns_2)
  
  return(columns_3)
}
```

```{r include=TRUE}
get_CT_average_expression_vec <- function(cluster_nums, av_expression_matrix, cluster_sizes) {
  # Given a vector of cluster numbers (which correspond to a given cell type),
  # we calculate average expression of each gene in the transcriptomic profile 
  # for that cell type.
  #
  # Args: 
  #   cluster_nums: vector of cluster numbers.
  #     e.g. c(12, 2, 20, 3, 0) correspond to ciona_larva_endo cell type.
  #
  #   av_expression_matrix: output of AverageExpression with "cluster flag" colnames (dataframe).
  #     e.g. AverageExpression(clean_chen_larva_integrated_20, features = final_khids)$RNA
  #
  #   cluster_sizes: table giving number of cells in each cluster.
  #     e.g. ciona_cluster_sizes <- table(Idents(clean_chen_larva_integrated_20))
  # 
  # Routine:
  #   1. Make "cluster flags" which will be used to select columns in av_expression_matrix.
  #     (columns_1)
  #   2. Get total expression of each gene in each cluster. (columns_2)
  #   3. Make 2 additional columns- total and average expression of each gene. 
  #     (columns_3)
  #   4. Take last (av_exp) column.
  #
  # Returns:
  #   CT_average_expression_vec: av_exp column in Step 4.
  cluster_flags <- make_cluster_flags(cluster_nums)
  
  columns_1 <- av_expression_matrix %>% select(all_of(cluster_flags))

  # contains both columns_2 and total_size
  columns_2_output <- make_columns_2(columns_1, cluster_nums, cluster_sizes)
  columns_2 <- columns_2_output[[1]]
  
  total_size <- columns_2_output[[2]]
  columns_3 <- make_columns_3(columns_2, total_size)
  
  CT_average_expression_vec <- columns_3$av_exp
  return(CT_average_expression_vec)
}

```

"""
Helper Function to make CT_vec_list 
(List with average vectors for each cell type)
"""
```{r include=TRUE}
make_CT_vec_list <- function(cluster_num_list, av_exp_matrix, cluster_sizes) {
  # Make a list in which each item corresponds to a Cell Type and average
  # expression of genes in the txnptomic profile.
  # Args:
  #     cluster_num_list: key: name of CT; value: its clusters
  #         e.g. Given ciona_heart_cluster_nums <- c(2, 1, 0), 
  #              list(heart = ciona_heart_cluster_nums)
  #         Note: Make this variable prior calling this function.
  # 
  #     av_exp_matrix: Will have already been made prior.
  #         Note: This will be the output of (e.g.)
  #               ciona_av_exp_matrix_1 <- make_av_exp_matrix(ciona_seurat_1,
  #                                                           ciona_txnptomic_profile)
  #
  #     cluster_sizes: Will have already been made prior. 
  #         Note: This will be the output of (e.g.)
  #               ciona_cluster_sizes_1 <- table(Idents(ciona_seurat_1))
  #
  # Returns:
  #     CT_vec_list: key: name of CT; value: Average Expression of genes 
  #                  in txnptomic profile for CT (vector).
  CT_vec_list <- list()
  CT_flags <- names(cluster_num_list)

  for (i in 1:length(cluster_num_list)) {
    CT_flag <- CT_flags[i]
    nums <- cluster_num_list[[i]]
    
    CT_vec <- get_CT_average_expression_vec(nums, av_exp_matrix, cluster_sizes)
    
    CT_vec_list[[CT_flag]] <- CT_vec
  }
  return(CT_vec_list)
}
```

"""
Helper Function to make Gene Specificity Matrix (Tosches)
"""
```{r include=TRUE}
make_gene_specificity_matrix <- function(average_expression_matrix) {
  # Given a average expression matrix, convert to a gene specificity matrix (Tosches)
  # Args:
  #   average_expression_matrix: 
  #       e.g. ciona_average_expression_matrix <- cbind(ciona_df_1, ciona_df_2)
  #
  # Routine:
  #     g_c/(1/N)sum(i in C)g_i
  #
  # Returns:
  #   gsm: Output of above routine.
  gsm_1 <- average_expression_matrix
  # makes new column with  mean of the row (average expression of gene across all cell types)
  gsm_1 <- gsm_1 %>% mutate(gene_CTs_avexp = rowMeans(.))
  num_cell_types <- ncol(gsm_1)-1
  gsm <- gsm_1[, 1:num_cell_types]/gsm_1[, num_cell_types+1]
  
  return(gsm)
}
```

""""""""""""""""""""""""""""""""
END HELPER FUNCTIONS
""""""""""""""""""""""""""""""""

"""
Module v3 Completed 7/2/20
"""

""""""""""""""""""""""""""""""""
       ** DO FOR CIONA ** 
""""""""""""""""""""""""""""""""

"""
Common Ciona Variables
"""

## Vary profile, Seurat Object ##
```{r include=FALSE}
ciona_txnptomic_profile <- final_khids

# All tissues except for heart + muscle
ciona_seurat_1 <- clean_chen_larva_integrated_20
ciona_cluster_sizes_1 <- table(Idents(ciona_seurat_1))
ciona_av_exp_matrix_1 <- make_av_exp_matrix(ciona_seurat_1, ciona_txnptomic_profile)

# Heart + Muscle
ciona_seurat_2 <- putative_hm_seurat
ciona_cluster_sizes_2 <- table(Idents(ciona_seurat_2))
ciona_av_exp_matrix_2 <- make_av_exp_matrix(ciona_seurat_2, ciona_txnptomic_profile)

```
"""
All Tissues
(ciona_cluster_num_list_1, ciona_CT_vec_list1)
"""

## This will vary depending on the Seurat Object used ##
```{r include=TRUE}
# 3274
ciona_endo_cluster_nums <- c(12, 2, 20, 3, 0)
# 1959
ciona_epi_cluster_nums <- c(28, 5, 25, 4, 18, 19)
# 4203
ciona_mes_cluster_nums <- c(7, 30, 13, 21, 23, 1, 11, 22, 8, 10, 35, 14, 6)
# 1033
ciona_cns_cluster_nums <- c(26, 9, 15, 24)
# 550
ciona_noto_cluster_nums <- c(27)

# Make this manually.
ciona_cluster_num_list_1 <- list(endo = ciona_endo_cluster_nums,
                               epi = ciona_epi_cluster_nums,
                               mes = ciona_mes_cluster_nums,
                               cns = ciona_cns_cluster_nums,
                               noto = ciona_noto_cluster_nums)
```

```{r include=TRUE}
ciona_CT_vec_list1 <- make_CT_vec_list(ciona_cluster_num_list_1, ciona_av_exp_matrix_1, ciona_cluster_sizes_1)
```

"""
Heart + Muscle
(ciona_cluster_num_list_2, ciona_CT_vec_list2)
"""
```{r include=TRUE}
# 207
ciona_heart_cluster_nums <- c(2, 1, 0)
# 159
ciona_mus_cluster_nums <- c(7, 4, 3, 6, 8)

ciona_cluster_num_list_2 <- list(heart = ciona_heart_cluster_nums,
                                 mus = ciona_mus_cluster_nums)
```

```{r include=TRUE}
ciona_CT_vec_list2 <- make_CT_vec_list(ciona_cluster_num_list_2, ciona_av_exp_matrix_2, ciona_cluster_sizes_2)
```

"""
Ciona Average Expression Matrix
(ciona_average_expression_matrix)
"""
# Do manually.
```{r include=TRUE}
ciona_df_1 <- as.data.frame(do.call(cbind, ciona_CT_vec_list1))
ciona_df_2 <- as.data.frame(do.call(cbind, ciona_CT_vec_list2))

ciona_average_expression_matrix <- cbind(ciona_df_1, ciona_df_2)
```

"""
Ciona Gene Specificity Matrix
(ciona_gsm)
"""
```{r include=TRUE}
ciona_gsm <- make_gene_specificity_matrix(ciona_average_expression_matrix)
rownames(ciona_gsm) <- rownames(ciona_average_expression_matrix)
```


""""""""""""""""""""""""""""""""
     ** DO FOR ZEBRAFISH ** 
""""""""""""""""""""""""""""""""

"""
Common Zeb Variables
"""
## Vary profile ##
```{r include=TRUE}
zeb_txnptomic_profile <- final_zeb_genes

# All tissues except for heart + muscle
zeb_seurat <- zeb_integrated_50
zeb_cluster_sizes <- table(Idents(zeb_seurat))
zeb_av_exp_matrix <- make_av_exp_matrix(zeb_seurat, zeb_txnptomic_profile)
```
"""
Select Zeb Tissues
(zeb_cluster_num_list, zeb_CT_vec_list)
"""
## Can Vary Cell Types ##
```{r include=TRUE}
# Sanity Checks (4)
zeb_noto_cluster_nums <- 7

zeb_heart_52_cluster_nums <- 74 
zeb_heart_51_56_cluster_nums <- 17 

zeb_muscles_cluster_nums <- 37

# Bulk tissue (2)
zeb_epi_1_cluster_nums <- c(52, 31, 65, 14)
zeb_epi_2_cluster_nums <- c(44, 45, 51) 
zeb_panc_prim_endo_cluster_nums <- 73
zeb_meso_cluster_names <- 46

# Should have low correlation with Ciona tissues... (2)
zeb_pec_fin_bud_cluster_nums <- 38
zeb_ph_arch_ndnf_cluster_nums <- 33 

# For fun (2)
zeb_iono_cluster_nums <- 64 
zeb_leuko_cluster_nums <- 67 

# Neural and Optic Cell Types (4)
zeb_lens_cluster_nums <- 57 
zeb_optic_cup_cluster_nums <- c(12, 7, 3, 43)
# Should be a "CNS Sanity Check"
zeb_diff_neus_cluster_nums <- c(9, 47, 59, 27, 16, 68, 41)
zeb_nc_grem2_cluster_nums <- 23

# make this yourself 
zeb_cluster_num_list <- list(noto = zeb_noto_cluster_nums,
                             heart_52 = zeb_heart_52_cluster_nums,
                             heart_51_56 = zeb_heart_51_56_cluster_nums,
                             muscles = zeb_muscles_cluster_nums,
                             epi_1 = zeb_epi_1_cluster_nums,
                             epi_2 = zeb_epi_2_cluster_nums,
                             panc_prim_endo = zeb_panc_prim_endo_cluster_nums,
                             meso = zeb_meso_cluster_names,
                             pec_fin_bud = zeb_pec_fin_bud_cluster_nums,
                             ph_arch_ndnf = zeb_ph_arch_ndnf_cluster_nums,
                             iono = zeb_iono_cluster_nums,
                             leuko = zeb_leuko_cluster_nums,
                             lens = zeb_lens_cluster_nums,
                             optic_cup = zeb_optic_cup_cluster_nums,
                             diff_neus = zeb_diff_neus_cluster_nums,
                             nc_grem2 = zeb_nc_grem2_cluster_nums)
```

```{r include=TRUE}
zeb_CT_vec_list <- make_CT_vec_list(zeb_cluster_num_list, zeb_av_exp_matrix, zeb_cluster_sizes)
```

"""
Zeb Average Expression Matrix
(zeb_average_expression_matrix)
"""

# Do manually.
```{r include=TRUE}
zeb_average_expression_matrix <- as.data.frame(do.call(cbind, zeb_CT_vec_list))
```

"""
Zeb Gene Specificity Matrix
(zeb_gsm)
"""
```{r include=TRUE}
zeb_gsm <- make_gene_specificity_matrix(zeb_average_expression_matrix)
rownames(zeb_gsm) <- rownames(zeb_average_expression_matrix)
```

"""""""""""""""""""""""""""
Do Spearman Correlations!
"""""""""""""""""""""""""""
# Helper Function 
```{r include=TRUE}
make_cor_df <- function(ciona_gsm, zeb_gsm) {
  cz_cor_df <- data.frame(ciona_CTs = 0,
                             zeb_CTs = 0,
                             cor = 0)
  
  for (i in 1:ncol(ciona_gsm)) {
    for (j in 1:ncol(zeb_gsm)) {
      # CT expression vectors 
      ciona_CT_col <- ciona_gsm[, i]
      zeb_CT_col <- zeb_gsm[, j]
      # CT name
      ciona_CT <- colnames(ciona_gsm)[i]
      zeb_CT <- colnames(zeb_gsm)[j]
      cat("i= ", i, "j= ", j, "ciona_CT= ", ciona_CT, "zeb_CT= ", zeb_CT, "\n")
      cz_cor <- cor(ciona_CT_col, zeb_CT_col, method = "spearman")
      cat("cor = ", cz_cor, "\n\n")
      cz_row <- c(ciona_CT, zeb_CT, cz_cor)
      cat("row = ", cz_row)
      cz_cor_df <- rbind(cz_cor_df, cz_row)
    }
  }
  cz_cor_df <- cz_cor_df[-1, ]
  return(cz_cor_df)
}
```

```{r include=TRUE}
cz_cor_df <- make_cor_df(ciona_gsm, zeb_gsm)
```

```{r include=TRUE}
cz_cor_df <- cz_cor_df %>% transform(cor = as.numeric(cor))
```

# Visualize
```{r include=TRUE}
# "RdYlBu", "PiYG"
cz_cor_heatmap <- ggplot(data = cz_cor_df,
                         aes(x=ciona_CTs, y=zeb_CTs, fill=cor)) +
                         geom_tile() +
                         scale_fill_distiller(palette = "RdBu")

cz_cor_heatmap
```




























"""
Just in case Module v3
"""
# Tests
# Test above function- seems correct! 
```{r include=TRUE}
t_aem <- ciona_average_expression_matrix
t_make_gsm <- make_gene_specificity_matrix(t_aem)
```

"""
Tested on Heart+Muscle- Correct~
test_CT_vec_list == ciona_CT_vec_list2
""" 
```{r include=TRUE}
test_CT_vec_list <- make_CT_vec_list(ciona_cluster_num_list_2, ciona_av_exp_matrix_2, ciona_cluster_sizes_2)
```


"""
Test everything!
Correct for Ciona Endo.
""" 
```{r include=FALSE}

test_obj <- clean_chen_larva_integrated_20
test_txnptomic_profile <- final_khids
test_cluster_nums <- c(12, 2, 20, 3, 0)
test_cluster_sizes <- table(Idents(test_obj))

test_av_exp_matrix <- make_av_exp_matrix(test_obj, test_txnptomic_profile)
```

```{r include=FALSE}
test_pipeline <- get_CT_average_expression_vec(test_cluster_nums, test_av_exp_matrix, test_cluster_sizes)
```

# Test- seems correct. 
```{r include=TRUE}
test_gsm_1 <- ciona_average_expression_matrix
# makes new column with  mean of the row (avgerage expression of gene across all cell types)
test_gsm_1 <- test_gsm_1 %>% mutate(gene_CTs_avexp = rowMeans(.))
num_cell_types <- ncol(test_gsm_1)-1
test_gsm_final <- test_gsm_1[, 1:num_cell_types]/test_gsm_1[, num_cell_types+1]
```

```{r include=TRUE}
# make this yourself 
# 3274
ciona_endo_cluster_nums <- c(12, 2, 20, 3, 0)
# 1959
ciona_epi_cluster_nums <- c(28, 5, 25, 4, 18, 19)
# 4203
ciona_mes_cluster_nums <- c(7, 30, 13, 21, 23, 1, 11, 22, 8, 10, 35, 14, 6)
# 1033
ciona_cns_cluster_nums <- c(26, 9, 15, 24)
# 550
ciona_noto_cluster_nums <- c(27)

# make this yourself 
ciona_cluster_num_list_1 <- list(endo = ciona_endo_cluster_nums,
                               epi = ciona_epi_cluster_nums,
                               mes = ciona_mes_cluster_nums,
                               cns = ciona_cns_cluster_nums,
                               noto = ciona_noto_cluster_nums)
```

# List of Average Vectors 
```{r include=TRUE}
ciona_flags_1 <- c("endo", "epi", "mes", "cns", "noto")
ciona_CT_vec_list1 <- list()

for (i in 1:length(ciona_cluster_num_list_1)) {
  ciona_flag_1 <- ciona_flags_1[[i]]
  ciona_nums_1 <- ciona_cluster_num_list_1[[i]]
  
  ciona_CT_vec_1 <- get_CT_average_expression_vec(ciona_nums_1, ciona_av_exp_matrix_1, ciona_cluster_sizes_1)
  
  ciona_CT_vec_list1[[ciona_flag_1]] <- ciona_CT_vec_1
}
```

"""
Old: Module v2 and Important Observations
"""

"""
Module v2: Do we get a signal if we use all orthologous genes? - WRONG IMPLEMENTATION
"""

# Question: Which 1-1 orthos are expressed in Ciona?
# Get KHID's for all orthos -WRONG
```{r include=TRUE}
all_khid_ortho_dir <- "/home/pprakriti/Git/Orthology-Maps/munging_code/all_khid_ortho_list.csv"

all_khid_ortho_df <- read.table(all_khid_ortho_dir,
                                sep = ",",
                                header = FALSE,
                                stringsAsFactors = FALSE)

all_khid_ortho_vec <- as.vector(all_khid_ortho_df$V1)
```

# Expression of 1-1 orthos in Ciona 
# Not all KHID orthos are expressed. -WRONG
```{r include=TRUE}
DoHeatmap(object = clean_chen_larva_integrated_20,
          features = all_khid_ortho_vec,
          cells = ciona_larva_noto,
          slot = "data",
          size = 2)
```

# Question: Which 1-1 orthos are expressed in Zeb?
# Get gene names of all orthos -WRONG
```{r include=TRUE}
all_zeb_gene_ortho_dir <- "/home/pprakriti/Git/Orthology-Maps/munging_code/all_zeb_gene_ortho_list.csv"

all_zeb_gene_ortho_df <- read.table(all_zeb_gene_ortho_dir,
                                sep = ",",
                                header = FALSE,
                                stringsAsFactors = FALSE)

all_zeb_gene_ortho_vec <- as.vector(all_zeb_gene_ortho_df$V1)
```

# Expression of 1-1 orthos in Zeb 
# Not all orthos are expressed. -WRONG
```{r include=TRUE}
# Note: there is a vector called zeb_ortho_gene_vec- the 6 DEGs in common
DoHeatmap(object = zeb_integrated_50,
          features = all_zeb_gene_ortho_vec,
          cells = noto,
          slot = "data",
          size = 2)
```

# How many 1-1 orthologs are expressed in each?
# I am looking at the intersection between the 2993 orthologs and the 2000 Var Features of each. -WRONG
```{r include=TRUE}
# 141
test_ciona_var_feats <- clean_chen_larva_integrated_20@assays$integrated@var.features
test_one_one_var_feats_ciona <- intersect(test_ciona_var_feats, all_khid_ortho_vec)

# 83
test_zeb_var_feats <- zeb_integrated_50@assays$integrated@var.features
test_one_one_var_feats_zeb <- intersect(test_zeb_var_feats, all_zeb_gene_ortho_vec)
```

# Now we want to know which 1-1 orthologs are expressed in both
# Write out 1-1 zeb gene orthologs (83)... -WRONG
```{r include=TRUE}
# This goes to my Git directory...
write.csv(x = test_one_one_var_feats_zeb,
          file = "test_one_one_var_feats_zeb.csv",
          row.names = FALSE)

```

# ... and get their KHIDs -WRONG
```{r include=TRUE}
# This is coming from scratch Desktop
khid_ortho_list_dir <- "/home/pprakriti/Desktop/temp_ortho/scratch_khid_ortho_list.csv"


khid_ortho_df <- read.table(khid_ortho_list_dir,
                                sep = ",",
                                header = FALSE,
                                stringsAsFactors = FALSE)

khid_ortho_vec <- as.vector(khid_ortho_df$V1)
```

# How many of these zeb orthologs (translated into KHIDs) are in expressed in Ciona? -WRONG
```{r include=TRUE}
# 25 :/ 
test_yes_khid <- intersect(test_ciona_var_feats, khid_ortho_vec)
```

# This is what the (25) profile looks like in Ciona. -WRONG
```{r include=TRUE}
DoHeatmap(object = clean_chen_larva_integrated_20,
          features = test_yes_khid,
          cells = ciona_larva_noto,
          slot = "data",
          size = 2)
```

# Take these KHIDs -> ENSC -> ENSD -> gene name -> visualize here. -WRONG
```{r include=TRUE}
write.csv(x = test_yes_khid,
          file = "test_yes_khid.csv",
          row.names = FALSE)
```

# These are the translated zeb gene names of above -WRONG
```{r include=TRUE}
zeb_25_ortho_list_dir <- "/home/pprakriti/Desktop/temp_ortho/scratch_25_zeb_gene_list.csv"


zeb_25_ortho_df <- read.table(zeb_25_ortho_list_dir,
                                sep = ",",
                                header = FALSE,
                                stringsAsFactors = FALSE)

zeb_25_ortho_vec <- as.vector(zeb_25_ortho_df$V1)
```

# This is what it looks like in Zeb -WRONG
```{r include=TRUE}
# Note: there is a vector called zeb_ortho_gene_vec- the 6 DEGs in common
DoHeatmap(object = zeb_integrated_50,
          features = zeb_25_ortho_vec,
          cells = noto,
          slot = "data",
          size = 2)
```

# Quick and dirty validation that the 25 genes are the same.
# They are not! Also the KHIDs are not 1-1 orthologs! 
```{r include=TRUE}

write.csv(x = test_yes_khid,
          file = "~/Desktop/test_yes_khid.csv",
          row.names = FALSE)
```

"""
Exploratory Questions
"""
# I want to get genes that aren't Var Genes- are they expressed in Ciona?

# Are all markers present in Var Genes? Try for notochord.
```{r include=TRUE}
test_noto_marker_df <- FindMarkers(clean_chen_larva_integrated_20,
                                 ident.1 = ciona_larva_noto) 
```

```{r include=TRUE}
# 762
test_noto_markers <- rownames(test_noto_marker_df)
test_ciona_var_genes <- clean_chen_larva_integrated_20@assays$integrated@var.features
```

# Conclusion: All markers are present in Var Genes. 
```{r include=TRUE}
# 762
test_intersect <- intersect(test_noto_markers, test_ciona_var_genes)
length(test_intersect == test_noto_markers)
```

# How about genes in data matrix? (14433)
```{r include=TRUE}
test_all_ciona_genes <- clean_chen_larva_integrated_20@assays$RNA@data@Dimnames[[1]]

# 12433
test_not_var_genes <- setdiff(test_all_ciona_genes, test_ciona_var_genes)
```

# Let's take the first 10 of these genes and check for expression.
```{r include=TRUE}
test_10_not_var_genes <- test_not_var_genes[1:10]
```

# These genes do not show up in 
```{r include=TRUE}
DoHeatmap(object = clean_chen_larva_integrated_20,
          features = test_10_not_var_genes,
          cells = ciona_larva_noto,
          slot = "data")
```




"""
TO DO
"""

# (Zeb) Are any of these in top50 DEGs for Zeb Noto?
```{r include=TRUE}
# No :(
zeb_true_vec <- zeb_ortho_gene_vec %in% ciona_noto_top50_markers
```


